src/main/java/org/spongepowered/api/item/inventory/generation/ItemStackBuilderPopulators.java

```diff
@@ -0,0 +1,197 @@
+            }
+        };
+    }
+
+    public static <E, V extends BaseValue<E>> BiConsumer<ItemStack.Builder, Random> value(V value) {
+        return (builder, random) -> {
+            final ItemStack itemStack = builder.build();
+            final DataTransactionResult dataTransactionResult = itemStack.offer(value);
+            if (dataTransactionResult.isSuccessful()) {
+                builder.from(itemStack);
+            }
+        };
+    }
+
+    public static BiConsumer<ItemStack.Builder, Random> data(DataManipulator<?, ?> manipulator) {
+        checkNotNull(manipulator, "DataManipulator cannot be null!");
+        return (builder, random) -> builder.itemData(manipulator);
+    }
+
+    public static BiConsumer<ItemStack.Builder, Random> data(Collection<DataManipulator<?, ?>> manipulators) {
+        checkNotNull(manipulators, "DataManipulators cannot be null!");
+        final WeightedTable<DataManipulator<?, ?>> table = new WeightedTable<>();
+        manipulators.forEach(manipulator -> table.add(checkNotNull(manipulator, "DataManipulator cannot be null!"), 1));
+        return (builder, random) -> builder.itemData(table.get(random).get(0));
+    }
+
+    public static BiConsumer<ItemStack.Builder, Random> data(Collection<DataManipulator<?, ?>> manipulators, VariableAmount rolls) {
+        checkNotNull(manipulators, "Manipulators cannot be null!");
+        checkNotNull(rolls, "VariableAmount cannot be null!");
+        final ImmutableList<DataManipulator<?, ?>> copied = ImmutableList.copyOf(manipulators);
+        return (builder, random) -> {
+            final int rollCount = rolls.getFlooredAmount(random);
+            checkArgument(rollCount >= 1, "Invalid VariableAmount! It must return at least 1!");
+            WeightedTable<DataManipulator<?, ?>> weightedTable = new WeightedTable<>(rollCount);
+            copied.forEach(manipulator1 -> weightedTable.add(manipulator1, 1));
+            weightedTable.get(random).forEach(builder::itemData);
+        };
+    }
+
+    public static BiConsumer<ItemStack.Builder, Random> data(WeightedTable<DataManipulator<?, ?>> weightedTable) {
+        checkNotNull(weightedTable, "WeightedTable cannot be null!");
+        return (builder, random) -> weightedTable.get(random).forEach(builder::itemData);
+    }
+
+    public static BiConsumer<ItemStack.Builder, Random> enchantment(Enchantment enchantment) {
+        return enchantment(VariableAmount.fixed(1), enchantment);
+    }
+
+    public static BiConsumer<ItemStack.Builder, Random> enchantment(VariableAmount level, Enchantment enchantment) {
+        checkNotNull(level, "VariableAmount cannot be null!");
+        checkNotNull(enchantment, "Enchantment cannot be null!");
+        return enchantments(VariableAmount.fixed(1), ImmutableList.of(new Tuple<>(enchantment, level)));
+    }
+
+    public static BiConsumer<ItemStack.Builder, Random> enchantmentsWithDefaultVariance(Collection<Enchantment> enchantments) {
```

Comment:
Probably needs a better name to coincide with what the method is actually doing.